core/DataAccess/Model.php

```diff
@@ -46,25 +46,20 @@ public function __construct(LoggerInterface $logger = null)
      * @return array
      * @throws Exception
      */
-    public function getInvalidatedArchiveIdsSafeToDelete($archiveTable, array $idSites)
+    public function getInvalidatedArchiveIdsSafeToDelete($archiveTable)
     {
         try {
             Db::get()->query('SET SESSION group_concat_max_len=' . (128 * 1024));
         } catch (\Exception $ex) {
             $this->logger->info("Could not set group_concat_max_len MySQL session variable.");
         }
 
-        $idSites = array_map(function ($v) { return (int)$v; }, $idSites);
-
         $sql = "SELECT idsite, date1, date2, period, name,
                        GROUP_CONCAT(idarchive, '.', value ORDER BY ts_archived DESC) as archives
                   FROM `$archiveTable`
                  WHERE name LIKE 'done%'
-                   AND value IN (" . ArchiveWriter::DONE_INVALIDATED . ','
-                                   . ArchiveWriter::DONE_OK . ','
-                                   . ArchiveWriter::DONE_OK_TEMPORARY . ")
-                   AND idsite IN (" . implode(',', $idSites) . ")
-                 GROUP BY idsite, date1, date2, period, name";
+                   AND `value` NOT IN (" . ArchiveWriter::DONE_ERROR . ")
```

Comment:
btw @diosmosis out of curiosity, do you maybe know why we don't delete an ERROR archive?

Reply:
I think it might be deleted somewhere else... I guess we can get rid of everything?

Reply:
Merged it for now @diosmosis . @mattab do you maybe remember? I reckon it's otherwise not too important

Reply:
Oh, I know what it's for, this is the SELECT, so we're looking for archives that are usable and deleting everything older. If we SELECT for DONE_ERROR archives, then the latest might be DONE_ERROR and we end up deleting the usable archives. We can include DONE_ERROR here but we'd have to a little extra processing below.

I think they are currently deleted in another task via `Model::getTemporaryArchivesOlderThan()`.

Reply:
makes sense, thanks 👍